tests/benchmark/bm_basics.cpp

```diff
@@ -171,29 +212,21 @@ BENCHMARK_DEFINE_F(BM_VecSimBasics, Range_BF)(benchmark::State &st) {
     float radius = (1.0f / 100.0f) * (float)st.range(0);
     size_t iter = 0;
     size_t total_res = 0;
+
     for (auto _ : st) {
-        st.PauseTiming();
-        // Generate random query vector before test.
-        std::uniform_real_distribution<double> distrib(-1.0, 1.0);
-        for (size_t i = 0; i < dim; ++i) {
-            query[i] = (float)distrib(rng);
-        }
-        st.ResumeTiming();
-        auto res = VecSimIndex_RangeQuery(bf_index, query.data(), radius, nullptr, BY_ID);
-        st.PauseTiming();
+        auto res =
+            VecSimIndex_RangeQuery(bf_index, (*queries)[iter++].data(), radius, nullptr, BY_ID);
```

Comment:
same goes here regarding generality, maybe add a module for the iteration to not overshoot the queries array